spec/integration/graphql_spec.rb

```diff
@@ -0,0 +1,176 @@
+
+        create_table :comments, force: true do |t|
+          t.text :body, null: false
+          t.belongs_to :post, null: false, index: true
+          t.timestamps
+        end
+      end
+
+      a = Post.create!(slug: 'a', title: 'A')
+      Post.create!(slug: 'b', title: 'B')
+      Post.create!(slug: 'c', title: 'C')
+      Comment.create!(post: a, body: 'So good')
+    end
+
+    let(:app) { Rails.application }
+
+    before :all do
+      module Types
+        class CommentType < GraphQL::Schema::Object
+          field :body, String, null: false
+        end
+
+        class PostType < GraphQL::Schema::Object
+          field :slug, String, null: false
+          field :title, String, null: false
+          field :comments, [CommentType], null: false
+        end
+
+        class QueryType < GraphQL::Schema::Object
+          field :posts, [PostType], null: false
+          field :post, PostType, null: false do
+            argument :slug, String, required: true
+          end
+
+          def posts
+            Post.all
+          end
+
+          def post(slug:)
+            Post.find_by(slug: slug)
+          end
+        end
+
+        class GraphQLTestAppSchema < GraphQL::Schema
+          query QueryType
+
+          use GraphQL::Execution::Interpreter
+          use GraphQL::Analysis::AST
+        end
+      end
+
+      class Post < ActiveRecord::Base
+        has_many :comments
+      end
+
+      class Comment < ActiveRecord::Base
+        belongs_to :post
+      end
+
+      setup_database
+
+      module GraphQLTestApp
+        class Application < Rails::Application
+          configure_rails_for_test
+
+          config.elastic_apm.disable_metrics = '*'
+          config.elastic_apm.api_request_time = '200ms'
+          config.logger = Logger.new(SpecLogger)
+        end
+      end
+
+      # rubocop:disable Style/ClassAndModuleChildren
+      class ::ApplicationController < ActionController::Base
+        def index
+          render plain: 'ok'
+        end
+      end
+
+      class ::GraphqlController < ApplicationController
+        def execute
+          variables = params[:variables]
+          query = params[:query]
+          operation_name = params[:operationName]
+          context = {
+            # Query context goes here, for example:
+            # current_user: current_user,
+          }
+          result = Types::GraphQLTestAppSchema.execute(
+            query,
+            variables: variables,
+            context: context,
+            operation_name: operation_name
+          )
+
+          render json: result
+        rescue => e
+          logger.error e.message
+          # logger.error e.backtrace.join("\n")
+
+          render(
+            status: 500,
+            json: {
+              error: { message: e.message, backtrace: e.backtrace },
+              data: {}
+            }
+          )
+        end
+      end
+      # rubocop:enable Style/ClassAndModuleChildren
+
+      MockIntake.stub!
```

Comment:
@estolfo I have to say, after the dust has settled on the spec suite metrics woes, I like the old `MockIntake` api better than the new, (imo) convoluted setup 😐

Reply:
I don't mind what we use, as long as it works 😄

Reply:
ps what is the "old API"? I lost track of the iterations.

Reply:
`MockIntake` _is_ the old API. _All the equivalent things_ we do in the Rails spec is the new API.